monai/transforms/io/array.py

```diff
@@ -129,3 +133,90 @@ def __call__(
+            - NIfTI files {``"bilinear"``, ``"nearest"``}
+                Interpolation mode to calculate output values.
+                See also: https://pytorch.org/docs/stable/nn.functional.html#grid-sample
+            - PNG files {``"nearest"``, ``"linear"``, ``"bilinear"``, ``"bicubic"``, ``"trilinear"``, ``"area"``}
+                The interpolation mode.
+                See also: https://pytorch.org/docs/stable/nn.functional.html#interpolate
+
+        padding_mode: This option is used when ``resample = True``. Defaults to ``"border"``.
+
+            - NIfTI files {``"zeros"``, ``"border"``, ``"reflection"``}
+                Padding mode for outside grid values.
+                See also: https://pytorch.org/docs/stable/nn.functional.html#grid-sample
+            - PNG files
+                This option is ignored.
+
+        scale: {``255``, ``65535``} postprocess data by clipping to [0, 1] and scaling
+            [0, 255] (uint8) or [0, 65535] (uint16). Default is None to disable scaling.
+            it's used for PNG format only.
+        dtype: data type during resampling computation. Defaults to ``np.float64`` for best precision.
+            if None, use the data type of input data. To be compatible with other modules,
+            the output data type is always ``np.float32``.
+            it's used for NIfTI format only.
+        output_dtype: data type for saving data. Defaults to ``np.float32``.
+            it's used for NIfTI format only.
+        save_batch: whether the import image is a batch data, default to `False`.
+            usually pre-transforms run for channel first data, while post-transforms run for batch data.
+
+    """
+
+    def __init__(
+        self,
+        output_dir: str = "./",
+        output_postfix: str = "trans",
+        output_ext: str = ".nii.gz",
+        resample: bool = True,
+        mode: Union[GridSampleMode, InterpolateMode, str] = "nearest",
+        padding_mode: Union[GridSamplePadMode, str] = GridSamplePadMode.BORDER,
+        scale: Optional[int] = None,
+        dtype: DtypeLike = np.float64,
+        output_dtype: DtypeLike = np.float32,
+        save_batch: bool = False,
+    ) -> None:
+        self.saver: Union[NiftiSaver, PNGSaver]
+        if output_ext in (".nii.gz", ".nii"):
+            self.saver = NiftiSaver(
+                output_dir=output_dir,
+                output_postfix=output_postfix,
+                output_ext=output_ext,
+                resample=resample,
+                mode=GridSampleMode(mode),
+                padding_mode=padding_mode,
+                dtype=dtype,
+                output_dtype=output_dtype,
+            )
+        elif output_ext == ".png":
+            self.saver = PNGSaver(
+                output_dir=output_dir,
+                output_postfix=output_postfix,
+                output_ext=output_ext,
+                resample=resample,
+                mode=InterpolateMode(mode),
+                scale=scale,
+            )
```

Comment:
else raise error?

Reply:
Sure, added in the latest commit.
Thanks.